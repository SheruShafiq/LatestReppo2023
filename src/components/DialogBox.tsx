import React from "react";
import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContentText,
  DialogTitle,
  SwipeableDrawer,
} from "@mui/material";
import useResizeHandler from "@/lib/hooks/useResizeHandler";

export type DialogBoxProps = {
  open: boolean;
  onContinue: () => void;
  onTerminate: () => void;
  title: string;
  description: string;
  confirmButtonText?: string;
  children?: React.ReactNode;
};

const DialogBox = ({
  open,
  children,
  onContinue,
  onTerminate,
  title,
  description,
  confirmButtonText,
}: DialogBoxProps) => {
  const size = useResizeHandler();

  return size.width > 600 ? (
    <Dialog
      open={open}
      sx={{
        "& .MuiDialog-paper": {
          width: "507px",
          minHeight: "254px",
          pt: "1rem",
          pr: "1rem",
          pl: "1rem",
        },
      }}
    >
      <DialogTitle minWidth={"427px"}>
        {title}
        <DialogContentText>{description}</DialogContentText>
      </DialogTitle>
      {children}
      <DialogActions sx={{ mt: "2rem", mr: "1rem" }}>
        <Button
          color="inherit"
          sx={{
            width: { xs: "22rem", sm: "8.5rem" },
            mr: { xs: 0, sm: "2rem" },
          }}
          onClick={onContinue}
        >
          Annuleren
        </Button>
        <Button
          variant="contained"
          sx={{ width: { xs: "22rem", sm: "14rem" } }}
          onClick={onTerminate}
        >
          {confirmButtonText ? confirmButtonText : "Beëindigen"}
        </Button>
      </DialogActions>
    </Dialog>
  ) : (
    <SwipeableDrawer
      anchor="bottom"
      open={open}
      onClose={() => {}}
      onOpen={() => console.log("open")}
      swipeAreaWidth={56}
      disableSwipeToOpen={false}
      sx={{
        "& .MuiDrawer-paper": {
          borderTopLeftRadius: 8,
          borderTopRightRadius: 8,
          p: "1.5rem",
        },
      }}
    >
      <Box sx={{ borderTopLeftRadius: 8, borderTopRightRadius: 8 }}>
        <DialogTitle
          sx={{
            padding: "0",
          }}
        >
          Weet je zeker dat je de mutatie wilt stoppen?
        </DialogTitle>
        <DialogContentText pt={"1rem"}>
          Als je de mutatie beëindigd dan gaan de gegevens die je hebt ingevoerd
          verloren.
        </DialogContentText>
        <DialogActions
          sx={{
            mt: "3.5rem",
            justifyContent: "space-around",
            padding: "0",
          }}
        >
          <Button
            color="inherit"
            sx={{
              flex: 1,
              fontSize: "0.8125rem",
            }}
            onClick={onContinue}
          >
            Annuleren
          </Button>
          <Button
            variant="contained"
            sx={{
              flex: 1,
              fontSize: "0.8125rem",
            }}
            onClick={onTerminate}
          >
            {confirmButtonText ? confirmButtonText : "Beëindigen"}
          </Button>
        </DialogActions>
      </Box>
    </SwipeableDrawer>
  );
};

export default DialogBox;
