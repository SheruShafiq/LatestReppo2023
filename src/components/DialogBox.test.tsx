// Require React Testing Library dependencies
import { render, fireEvent } from "@testing-library/react";
import DialogBox from "./DialogBox";
import matchers from "@testing-library/jest-dom";
import { debug } from "vitest-preview";
expect.extend(matchers);

describe("DialogBox Component", () => {
  it("renders successfully", () => {
    const { getByText } = render(
      <DialogBox
        title="test title"
        description="test description"
        onContinue={vi.fn()}
        onTerminate={vi.fn()}
        open={true}
      />
    );

    expect(getByText("test title")).toBeTruthy();
    expect(getByText("test description")).toBeTruthy();
  });

  it("renders title and description correctly", () => {
    const { getByText } = render(
      <DialogBox
        title="test title"
        description="test description"
        onContinue={vi.fn()}
        onTerminate={vi.fn()}
        open={true}
      />
    );

    expect(getByText("test title")).toBeInTheDocument();
    expect(getByText("test description")).toBeInTheDocument();
  });

  it("onContinue is called on button click", () => {
    const onContinueMock = vi.fn();

    const { getByText } = render(
      <DialogBox
        title="test title"
        description="test description"
        onContinue={onContinueMock}
        onTerminate={() => {}}
        open={true}
      />
    );

    fireEvent.click(getByText("Annuleren"));
    expect(onContinueMock).toHaveBeenCalledTimes(1);
  });

  it("onTerminate is called when Terminate button is clicked", () => {
    const onTerminateMock = vi.fn();

    const { getByText } = render(
      <DialogBox
        open={true}
        title="test title"
        description="test description"
        onContinue={() => {}}
        onTerminate={onTerminateMock}
      />
    );

    fireEvent.click(getByText("Beëindigen"));
    expect(onTerminateMock).toHaveBeenCalledTimes(1);
  });

  it("renders Terminate button text correctly", () => {
    // Default text
    let { getByText } = render(
      <DialogBox
        title="test title"
        description="test description"
        onContinue={vi.fn()}
        onTerminate={vi.fn()}
        open={true}
      />
    );

    expect(getByText("Beëindigen")).toBeInTheDocument();

    // Custom text
    render(
      <DialogBox
        open={true}
        title="test title"
        description="test description"
        onContinue={vi.fn()}
        onTerminate={vi.fn()}
        confirmButtonText="custom text"
      />
    );

    expect(getByText("custom text")).toBeInTheDocument();
    expect(screen).toMatchSnapshot();
  });
});
