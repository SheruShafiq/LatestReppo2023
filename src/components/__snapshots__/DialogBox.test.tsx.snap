// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DialogBox Component > renders Terminate button text correctly 1`] = `Screen {}`;
